roles/jd-server/src/lib/job_declarator/mod.rs

```diff
@@ -193,6 +312,16 @@ impl JobDeclaratorDownstream {
                         break;
                     }
                 }
+                let retrieve_transactions =
+                    JobDeclaratorDownstream::retrieve_transactions_via_rpc(self_mutex.clone())
+                        .await;
```

Comment:
why you are awaiting it? the all point of this PR is to get tx data without blocking here you just block the main loop. This should run in a new task and not be awaited.

Reply:
This has been fixed by @lorbax in https://github.com/GitGab19/stratum/pull/15/commits/b63587037769b5cc12dfad6108cafe2db4b88b14